### PR TITLE
Don't use or store id_token when using cookie-based auth

### PR DESCRIPTION
**Pull request by absoludity**
This is a prequel to re-enabling the logout option for oidc. When I first re-enabled logout, I found that the cookie would be removed as would the id_token (from localStorage), but then the client would just re-get the id_token from the server (`fetchOIDCToken`) and therefore assume it was logged in again.

One of the benefits of using OAuth/OIDC is that the client doesn't need to ever see (or store) the actual access token... the http-only cookie is all that is required (and the auth-proxy adds the id-token to the request), which is not only much safer, but means we can logout also :)

### CODE REVIEW

File: dashboard/src/shared/Auth.ts [hunk 1]
```diff
@@ -65,21 +68,23 @@ export class Auth {
...
+  public static async isAuthenticatedWithCookie(): Promise<boolean> {
     try {
-      const { headers } = await Axios.head("");
-      if (headers && headers.authorization) {
-        const tokenMatch = (headers.authorization as string).match(/Bearer\s(.*)/);
-        if (tokenMatch) {
-          return tokenMatch[1];
-        }
-      }
+      await Axios.get(APIBase + "/");
```
Comment (andresmgot) [inline dashboard/src/shared/Auth.ts]: this should be `Axios.head`, no? (or maybe the comment is wrong)

File: dashboard/src/shared/Auth.test.ts [hunk 2]
```diff
@@ -58,29 +59,42 @@ describe("Auth", () => {
...
 
-  describe("fetchOIDCToken", () => {
-    it("should fetch a token", async () => {
-      Axios.head = jest.fn(path => {
-        expect(path).toEqual("");
-        return { headers: { authorization: "Bearer foo" } };
+  describe("isAuthenticatedWithCookie", () => {
+    it("returns true if request to API root succeeds", async () => {
+      Axios.get = jest.fn(path => {
+        return { headers: { status: 200 } };
```
Comment (andresmgot) [inline dashboard/src/shared/Auth.test.ts]: nit: this should probably return a promise too (to match the other tests)

Comment (absoludity) [reply dashboard/src/shared/Auth.test.ts]: Updated.

File: dashboard/src/shared/Auth.ts (hunk 1 above)
Comment (absoludity) [reply dashboard/src/shared/Auth.ts]: The comment is wrong (need to check the body of the error response), but already fixed in the following branch.

**Pull request merged**